### PR TITLE
Integrate CollectOrderPaymentUseCase on Order Details

### PR DESCRIPTION
**Pull request by Ecarrion**
Part of #5655 

# Why

`CollectOrderPaymentUseCase` was created by abstracting the payments code from `OrderDetailsViewModel` and `OrderDetailsViewController` into a single place with the intention that it could be reused for both flows.

`CollectOrderPaymentUseCase` was integrated on Simple Payments on https://github.com/woocommerce/woocommerce-ios/pull/5581 but it wasn't integrated on `Order Details` in order to not include any regression and be able to test "Order Details" in isolation.

This PR integrates `CollectOrderPaymentUseCase` on `Order Details` and removes the unnecessary code.

# Demo

https://user-images.githubusercontent.com/562080/147793987-dcb2f469-619c-4c0b-ab1d-031706da4929.MP4

https://user-images.githubusercontent.com/562080/147794002-e4218c75-247c-4500-841b-d3298096deb6.MP4

# Testing Instructions

- Test the order details collect payment flow & edge cases

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by allendav** [comment]
Will look at this asap

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift [hunk 1]
```diff
@@ -519,66 +503,30 @@ extension OrderDetailsViewModel {
...
-                        onProcessingMessage: @escaping () -> Void, // i.e. payment is processing
-                        onDisplayMessage: @escaping (String) -> Void, // e.g. "Remove Card"
-                        onCompletion: @escaping (Result<CardPresentReceiptParameters, Error>) -> Void) { // used to tell user payment completed (or not)
-        /// We don't have a concept of priority yet, so use the first paymentGatewayAccount for now
-        /// since we can't yet have multiple accounts
-        ///
-        if self.cardPresentPaymentGatewayAccounts.count != 1 {
-            DDLogWarn("Expected one card present gateway account. Got something else.")
+    /// Collects payments for the current order.
+    /// Tries to connect to a reader if necesary.
```
Comment (allendav) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift]: necessary

**Review by allendav — APPROVED** [review]

**Comment by Ecarrion** [comment]
Thanks for testing @allendav 🙏
